EOFM intro and evaluation from draft servir book chapter

lillythomas opened the pull request:
Here is a copy of the notebook that we made for NASA SERVIR's applied deep learning science book. We can use/adjust this in any way we see fit. 

A requirements file for the notebook is included in this PR but would be good to discuss how we might want to incorporate those libraries into the jupyter book build @weiji14.

lillythomas committed:
chapter from servir

weiji14 committed:
Migrate geospatial requirements from requirements.txt to pyproject.toml

Specifically geopandas, rioxarray and stackstac. Not migrating dependencies which are sub-dependencies of those already specified, or those unused in the jupyter notebook.

weiji14 committed:
Migrate ml requirements from requirements.txt to pyproject.toml

Specifically einops, huggingface-hub, scikit-learn, terratorch, torch and torchvision.

weiji14 committed:
Gitignore .ipynb_checkpoints/

weiji14 commented on .github/workflows/deploy.yml:
```diff
@@ -5,7 +5,7 @@ name: MyST GitHub Pages Deploy
 on:
   # Uncomment the 'pull_request' line below to manually re-build Jupyter Book
-  # pull_request:
+  pull_request:
```
```suggestion
  # pull_request:
```
TODO comment out this line again before merging.

weiji14 reviewed (commented):
> Here is a copy of the notebook that we made for NASA SERVIR's applied deep learning science book. We can use/adjust this in any way we see fit.
>
> A requirements file for the notebook is included in this PR but would be good to discuss how we might want to incorporate those libraries into the jupyter book build @weiji14.

Awesome, thanks for copying this over @lillythomas! I've migrated the dependencies from requirements.txt to the pyproject.toml file (used by `uv`). Will be good to test how this might work on Google Colab or Sagemaker Studio Lab.

For the Jupyter Book build, I've temporarily triggered a deployment for this branch. You can preview it at https://developmentseed.org/igarss25tutorial/tut2-eofm-evaluation, or by running `uv run jupyter book start` locally.

weiji14 committed:
Put tut2_EOFM_Evaluation.ipynb in table of contents

weiji14 committed:
Temporarily trigger jupyter book build

lillythomas committed:
updates for changed clay module structure

weiji14 committed:
Move overview text to tut1_intro.md

weiji14 committed:
Use DOI links to generate citation list automatically

Xref https://next.jupyterbook.org/tutorial/mystmd/#citations

weiji14 committed:
Add acknowledgements for SERVIR

Thanks for allowing us to re-use some content from a draft book chapter of https://github.com/SERVIR/SERVIR-Applied-Deep-Learning-Book

weiji14 committed:
Revert accidental uv commands

weiji14 committed:
Revert "Temporarily trigger jupyter book build"

This reverts commit 50bbdd71c03c8b5d0edb53db38c528257948f068.

weiji14 reviewed (approved):
Thanks @lillythomas, ok if we merge this as is, and then continue to polish things up on @srmsoumya's branch at #6?

lillythomas commented:
Sounds good, thanks for the help on this one @weiji14 !
